nix/install.in

```diff
@@ -5,19 +5,22 @@
 # (which in turn creates and populates /nix).
 
 { # Prevent execution if this script was only partially downloaded
-
-unpack=nix-binary-tarball-unpack
+oops() {
+    echo "$0: $@" >&2
+    exit 1
+}
+unpack="$(mktemp -d nix-binary-tarball-unpack.XXXXXXXXXX || \
+          oops "Can't create tempdir for unpacking")"
+cleanup() {
+    rm -rf "$unpack"
+}
+trap cleanup EXIT
```

Comment:
Yes, unfortunately there are no real standards here. We have to do our best and cover 90% of the cases. Thanks!